op-service/txmgr/txmgr.go

```diff
@@ -101,6 +121,102 @@ type SimpleTxManager struct {
 	l       log.Logger
 }
 
+// TxCandidate is a transaction candidate that can be submitted to ask the
+// [TxManager] to construct a transaction with gas price bounds.
+type TxCandidate struct {
+	// TxData is the transaction data to be used in the constructed tx.
+	TxData []byte
+	// Recipient is the recipient (or `to`) of the constructed tx.
+	Recipient common.Address
+	// GasLimit is the gas limit to be used in the constructed tx.
+	GasLimit uint64
+	// From is the sender (or `from`) of the constructed tx.
+	From common.Address
+	/// ChainID is the chain ID to be used in the constructed tx.
+	ChainID *big.Int
```

Comment:
Should probably been in the `TxManager` constructor (or fetched from L1 by the txmgr) rather than in the `TxCandidate`